### PR TITLE
[CodeStyle][C416][C417] rewrite unnecessary comprehension with function call and use generator instead of map

### PR DESCRIPTION
**Pull request by enkilee**
### PR types

Others
### PR changes

Others
### Describe

- https://github.com/PaddlePaddle/Paddle/issues/51729
case 31 case 32
### 1
modify pyproject.toml:
add C413 C414 C416 C417

### 2
According https://pypi.org/project/flake8-comprehensions/

C416: Unnecessary <dict/list/set> comprehension - rewrite using <dict/list/set>().
It’s unnecessary to use a dict/list/set comprehension to build a data structure if the elements are unchanged. Wrap the iterable with dict(), list(), or set() instead. For example:

Rewrite {a: b for a, b in iterable} as dict(iterable)
Rewrite [x for x in iterable] as list(iterable)
Rewrite {x for x in iterable} as set(iterable)

C417: Unnecessary map usage - rewrite using a generator expression/<list/set/dict> comprehension.
map(func, iterable) has great performance when func is a built-in function, and it makes sense if your function already has a name. But if your func is a lambda, it’s faster to use a generator expression or a comprehension, as it avoids the function call overhead. For example:

Rewrite map(lambda x: x + 1, iterable) to (x + 1 for x in iterable)
Rewrite map(lambda item: get_id(item), items) to (get_id(item) for item in items)
Rewrite list(map(lambda num: num * 2, nums)) to [num * 2 for num in nums]
Rewrite set(map(lambda num: num % 2 == 0, nums)) to {num % 2 == 0 for num in nums}
Rewrite dict(map(lambda v: (v, v ** 2), values)) to {v : v ** 2 for v in values}

In C416:
```python
>>>timeit("{x for x in range(1,100)}")
2.8943359665572643
>>>timeit("set(range(1,100))")
1.5967901349067688
```

In C417:
```python
>>>timeit("list(map(lambda num: num * 2, range(1,10)))")
1.117645412683487
>>>timeit("[num * 2 for num in range(1,10)]")
0.638909362256527
```

Two rules all can use it.
using rule：✅
auto fix：✅

command as follow：

```python
# install ruff 0.0.254
pip install ruff==0.0.254
# using command to auto fix:
ruff --select C416 . --fix
ruff --select C417 . --fix
```

### CODE REVIEW

**Comment by SigureMo** [comment]
> **Warning**
>
> 由于配置关系，本 PR 应在下列 PR merge 后 merge
> - #52065

应该在显眼处说明 PR 的先后 merge 关系

File: pyproject.toml [hunk 1]
```diff
@@ -39,12 +39,12 @@ select = [
...
     "C410",
-    "C411",
-    # "C413",
-    # "C414",
+    # "C411",
+    "C413",
+    "C414",
     # "C415",
-    # "C416",
-    # "C417",
```
Comment (SigureMo) [inline pyproject.toml:50]: 这里是解决冲突出错了嘛？，`C411` 怎么注释掉了

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_loop.py [hunk 2]
```diff
@@ -222,9 +222,9 @@ def for_loop_dufunc_with_listcomp(array):
     a = 1
     for j in range(array):
         res = [x + a for x in array]
-        res = [i for i in array]
+        res = list(array)
         x = 1
-    b = [i for i in array]
+    b = list(array)
```
Comment (SigureMo) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_loop.py]: 根据函数名，这里就是用来测试 list comprehension 的，需要回退修改并在配置中 `tool.ruff.per-file-ignores` ignore 掉本文件

**Review by SigureMo — COMMENTED** [review]
不建议这种相似性不高的合在一起，不过这个 PR 就别拆了

另外需要解决下冲突

**Review by SigureMo — APPROVED** [review]
LGTM

我在这个 PR 里整理了下配置，并解决了一些增量问题，以收尾 C4 系列

**Comment by enkilee** [comment]
> LGTM
> 
> 我在这个 PR 里整理了下配置，并解决了一些增量问题，以收尾 C4 系列

辛苦大佬♥

**Pull request merged**